src/test/java/seedu/address/logic/commands/SummaryCommandTest.java

```diff
@@ -0,0 +1,169 @@
+        Person highest = summaryCommand.getHighest(lastShownList);
+        assertNotNull(highest);
+        assertEquals(BENSON, highest);
+    }
+
+    @Test
+    void getLowest_returnsPersonWithLowestCommission() {
+        List<Person> lastShownList = List.of(ALICE, BENSON);
+        SummaryCommand summaryCommand = new SummaryCommand();
+        Person lowest = summaryCommand.getLowest(lastShownList);
+        // ALICE should have the lowest commission ("100" < "200")
+        assertNotNull(lowest);
+        assertEquals(ALICE, lowest);
+    }
+
+    @Test
+    void calculateAverageCommission_correctlyAveragesValues() {
+        List<Person> lastShownList = List.of(ALICE, BENSON);
+        SummaryCommand summaryCommand = new SummaryCommand();
+        double averageCommission = summaryCommand.getAverage(lastShownList);
+
+        double expectedAverage = (Integer.parseInt(ALICE.getCommission().value)
+                + Integer.parseInt(BENSON.getCommission().value)) / 2.0;
+        // Allow a small delta for floating point comparison
+        assertEquals(expectedAverage, averageCommission, 0.0001);
+    }
+
+    @Test
+    public void execute_summaryWithTypicalPersons_success() {
+        // Arrange
+        SummaryCommand summaryCommand = new SummaryCommand(); // Use correct command name
+
+        // --- Calculate Expected Values Manually Based on TypicalPersons ---
+        List<Person> personList = model.getFilteredPersonList();
+        BigDecimal totalCommissionValue = BigDecimal.ZERO;
+        Person highestCommissionPerson = null;
+        Person lowestCommissionPerson = null;
+        BigDecimal highestCommission = BigDecimal.valueOf(Double.MIN_VALUE);
+        // Initialize lowestCommission with a very large value or the first person's commission
+        BigDecimal lowestCommission = null;
+
+        for (Person person : personList) {
+            BigDecimal currentCommission = new BigDecimal(person.getCommission().value);
+            totalCommissionValue = totalCommissionValue.add(currentCommission);
+
+            if (currentCommission.compareTo(highestCommission) > 0) {
+                highestCommission = currentCommission;
+                highestCommissionPerson = person;
+            }
+            // Initialize lowest on first iteration or compare
+            if (lowestCommission == null || currentCommission.compareTo(lowestCommission) < 0) {
+                lowestCommission = currentCommission;
+                lowestCommissionPerson = person;
+            }
+        }
+
+        BigDecimal averageCommissionValue = BigDecimal.ZERO;
+        if (!personList.isEmpty()) {
+            averageCommissionValue = totalCommissionValue.divide(
+                    new BigDecimal(personList.size()), 2, RoundingMode.HALF_UP);
+        }
+        // --- End Calculation ---
+
+
+        // --- Construct Expected Feedback Message ---
+        String expectedStatsMessage = "==== General Statistics ====\n"
```

Comment:
If you package Summary into a class, you can add a static non-public constructor with params being the raw stats and call the toString method?